### ISSUE TITLE
[TB]-"Fix Hex Value Visibility and Dynamic Color Name Display in Color Palette Component"

### ISSUE THREAD

**Issue by AradhyaDixit18** [finos/a11y-theme-builder#946]
The current implementation of the ColorPaletteAtom component has two major issues:

1)Hex Value Visibility: The hex value of the selected color is displayed in black, causing it to become invisible against dark colors.

2)Dynamic Color Name Display: When a color is selected using the color picker, the color name field does not update to display the name of the selected color.

**Proposed Changes/Solution:**

Separate Color Display Block: Add a block to visually display the selected color, instead of showing the color in the hex value field.
Hex Value TextField: Keep the hex value in the TextField without changing its background color to ensure visibility.
Dynamic Color Name Update: Update the color name field to correctly display the name of the selected color.
This issue aims to improve the user interface and usability of the color palette component.

**Comment by AradhyaDixit18** [comment]
@aaronreed708 can you please assign this task to me as I have already fixed the issue and created the PR.

**Comment by AradhyaDixit18** [comment]
[enhancement]

**Comment by aaronreed708** [comment]
This is an example of the issue.  Notice that the hex value of the selected color is not visible in the red rectangle.

<img width="545" alt="Screenshot 2024-07-08 at 10 58 32 PM" src="https://github.com/finos/a11y-theme-builder/assets/32913837/2ff66c12-121c-4b93-8036-e181f01f1ca7">

**Comment by aaronreed708** [comment]
Hi @AradhyaDixit18!  Thank you for opening this issue and making your contribution!  This is certainly an issue that needs to be fixed.  However, I'm wondering if this is the best solution.  The palette is already two fields, I don't know if having a third field taking up more space is the most efficient solution.

There is also already code in ColorPaletteAtom.reflectColorPickerChangeInUI():  https://github.com/finos/a11y-theme-builder/blob/0f3e3938c00fa56b0fa26ff88206985f5dc0b724/code/src/ui/src/pages/atoms/ColorPaletteAtom.tsx#L51-L61

whose job it is to change the text color to contrast with the selected background color.  But it appears to not be working.  If you have time, could you take a look at why it isn't working?

**Comment by aaronreed708** [comment]
Also, you mentioned that one of the issues is:

_2)Dynamic Color Name Display: When a color is selected using the color picker, the color name field does not update to display the name of the selected color._

I kinda like the change that you made to generate a color name when the user selects the color.  But one of the issues is that this color name field should be able to hold any name that I give it.  Think of it as not being a literal "color name", but the alias that I am giving this color value.  For example, I could call a color "Midnight Mist" or "Brick" or "Logo Background" or "1958 Logo".  It might depend on the color's relationship with the design system or just a random value that I gave it.

So in your implementation, if I set the value of that field, then select the color, then the name that I gave the color is overwritten which I feel is not a great user experience.  Perhaps we could use your generated color name as a suggestion or a placeholder value if the user hasn't already provided a name.  But it probably shouldn't replace what the user has entered.
